diff --git a/src/tty_interface.c b/src/tty_interface.c
index db0b7e2..5c8c3dd 100644
--- a/src/tty_interface.c
+++ b/src/tty_interface.c
@@ -228,6 +228,7 @@ static const keybinding_t keybindings[] = {{"\x7f", action_del_char},	/* DEL */
 					   {KEY_CTRL('P'), action_prev},	 /* C-P */
 					   {KEY_CTRL('N'), action_next},	 /* C-N */
 
+					   {"\x1b",   action_exit}, /* ESC */
 					   {"\x1b[A", action_prev}, /* UP */
 					   {"\x1bOA", action_prev}, /* UP */
 					   {"\x1b[B", action_next}, /* DOWN */
