From 2465698574c921c86565fbcc496e4f8657a2ab7d Mon Sep 17 00:00:00 2001
From: Georgy Yakovlev <gyakovlev@gentoo.org>
Date: Mon, 17 Feb 2020 14:15:54 -0800
Subject: [PATCH] add dtinfo ppc defines

---
 config/cf/DtInfo.tmpl | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/config/cf/DtInfo.tmpl b/config/cf/DtInfo.tmpl
index 3a5eb056..e9ce99f9 100644
--- a/config/cf/DtInfo.tmpl
+++ b/config/cf/DtInfo.tmpl
@@ -416,6 +416,12 @@ TEMPLATE_OBJS=TemplateDB//**/*.o
 # define ByteOrderDefines LittleEndianDefines
 #elif defined(HAL32V7Architecture) || defined(SunArchitecture) || defined(AIXArchitecture) || defined(HPArchitecture)
 # define ByteOrderDefines BigEndianDefines
+#elif defined(PpcArchitecture)
+# if defined(__LITTLE_ENDIAN__)
+#  define ByteOrderDefines LittleEndianDefines
+#else
+#  define ByteOrderDefines BigEndianDefines
+# endif
 #else
 # define ByteOrderDefines DefaultEndianDefines
 #endif
-- 
2.25.1

